libp2p/protocols/pubsub/pubsub.nim

```diff
@@ -80,10 +80,27 @@ method subscribeTopic*(p: PubSub,
                        topic: string,
                        subscribe: bool,
                        peerId: string) {.base, async.} =
+  var peer = p.peers.getOrDefault(peerId)
+  if isNil(peer) or isNil(peer.peerInfo): # should not happen
+    if subscribe:
+      warn "subscribeTopic but peer was unknown!"
+      return # Stop causing bad metrics!
+    else:
+      return # Stop causing bad metrics!
+  
+  let idx = peer.topics.find(topic)
   if subscribe:
     libp2p_pubsub_peers_per_topic.inc(labelValues = [topic])
```

Comment:
IMO, this is a bit excessive and there isn't a good way to do this generically. For one, each protocol has it's own subscriptions, for example floodsub and gossipsub have separate tables. I was going to remove this metric altogether and I still think we should because we're not going to get an accurate view. In gossipsub, we already capture it under `libp2p_gossipsub_peers_per_topic_gossipsub` anyways.

Reply:
Fine by me to remove it, not really necessary indeed.

Reply:
The only doubt here I have is that `peer.topics` won't get populated properly tho

Reply:
The ` topics*: seq[string]` I mean in `PubSubPeer`

Reply:
It is in fact not used anywhere... I will remove it